src/components/RequestsTable/RequestsTable.tsx

```diff
@@ -125,11 +125,22 @@ const CurrentStage = createTableColumn<RPARequest>({
     return <>Current Stage {filtered && <FilterIcon />}</>;
   },
   renderCell: (item) => {
-    return (
-      <TableCellLayout truncate>
-        {STAGES.filter((stage) => stage.key === item.stage)[0].text}
-      </TableCellLayout>
+    const stageIndex = STAGES.findIndex(({ key }) => key === item.stage);
+    const currentStage = STAGES[stageIndex]?.text;
+
+    const subStageIndex = STAGES[stageIndex].subStages?.findIndex(
+      ({ key }) => key === item.subStage
     );
+    let subStage = undefined;
+    if (subStageIndex !== undefined) {
+      let subStages = STAGES[stageIndex].subStages;
+      if (subStages) {
+        subStage = subStages[subStageIndex]?.text;
+      }
+    }
+
+    let displayStage = subStage || currentStage;
+    return <TableCellLayout truncate>{displayStage}</TableCellLayout>;
```

Comment:
Since you are only using subStageIndex to get the text, why not just go straight to get the text rather than getting an index, then getting the text?

```suggestion
    const subStage = STAGES[stageIndex].subStages?.find(
      ({ key }) => key === item.subStage
    )?.text;
    
    let displayStage = subStage || currentStage;
    return <TableCellLayout truncate>{displayStage}</TableCellLayout>;
```